Objects/typeobject.c

```diff
@@ -287,32 +290,45 @@ type_mro_modified(PyTypeObject *type, PyObject *bases) {
 
        Unset HAVE_VERSION_TAG and VALID_VERSION_TAG if the type
        has a custom MRO that includes a type which is not officially
-       super type.
+       super type, or if the type implements its own mro() method.
 
        Called from mro_internal, which will subsequently be called on
        each subclass when their mro is recursively updated.
      */
     Py_ssize_t i, n;
     int clear = 0;
+    int custom = (Py_TYPE(type) != &PyType_Type);
 
     if (!PyType_HasFeature(type, Py_TPFLAGS_HAVE_VERSION_TAG))
         return;
 
-    n = PyTuple_GET_SIZE(bases);
-    for (i = 0; i < n; i++) {
-        PyObject *b = PyTuple_GET_ITEM(bases, i);
-        PyTypeObject *cls;
+    if (custom) {
+        _Py_IDENTIFIER(mro);
+        int unbound;
+        PyObject *mro_meth = lookup_method((PyObject *)type, &PyId_mro,
```

Comment:
This first `lookup_method()` call should be checked for failure before the second call. If the first call fails, an assertion failure may occur during the second call in `_PyType_Lookup()` (as `_PyType_Lookup()` performs `assert(!PyErr_Occurred());`).

Reply:
Thanks for the review @ZackerySpytz :) It's fixed, I think, by avoiding to call lookup_method which is the one setting the error.